src/talkaction.cpp

```diff
@@ -61,7 +61,9 @@ Event* TalkActions::getEvent(const std::string& nodeName)
 
 bool TalkActions::registerEvent(Event* event, const pugi::xml_node&)
 {
-	talkActions.push_front(*static_cast<TalkAction*>(event)); // event is guaranteed to be a TalkAction
+	TalkAction* talkAction = static_cast<TalkAction*>(event); // event is guaranteed to be a TalkAction
+	talkActions.push_front(*talkAction);
```

Comment:
`std::move(*talkaction)` might use less resources than copying here. Also I'd use an `unique_ptr` for clean pointer handling.

```
auto talkaction = std::unique_ptr<Talkaction>(*event);
talkactions.push_front(std::move(*talkaction.get()));
```